src/pds/updart/client.py

```diff
@@ -94,7 +135,7 @@ def __next__(self):
         :return: next filtered product
         """
         try:
-            return next(self._crt_page_iterator)
+            return self._get_new_product()
         except StopIteration:
-            # TODO: implement next page retrieval and return first element of next page
-            raise
+            self._init_new_page()
+            return self._get_new_product()
```

Comment:
This whole `_init_new_page`, `return _get_new_product`, and `try…except StopIteration` could be simplified to fewer lines using a generator with `yield`

Reply:
Thanks @nutjob4life , I don't see immediately how that would work. We can go over that during a breakout session, I will schedule it for this Tuesday.

Reply:
Oh no! I should've kept my mouth shut! 😂